src/_pytest/main.py

```diff
@@ -423,6 +450,10 @@ def pytest_ignore_collect(collection_path: Path, config: Config) -> bool | None:
         if any(fnmatch_ex(pat, collection_path) for pat in norecursepatterns):
             return True
 
+        if any(fnmatch_ex(pat, collection_path) for pat in ("build", "dist")):
+            if _is_setuptools_project(collection_path.parent):
```

Comment:
This case does not need a nested block. Instead, both conditions could be merged using `and`. The line would be quite long, though, so it might be a good idea to assign that to a variable that would also be an opportunity to assign semantics to this check through naming the var with something that corresponds to the high-level purpose of the check.

For example, using `is_distributable_library` would make that check `if is_distributable_library:`, which is both short and communicates what the list of low-level checks mean.

Reply:
tried an `allow_in_build` approach, let me know what you think